dpnp/dpnp_array.py

```diff
@@ -347,22 +347,22 @@ def __gt__(self, other):
 
     def __iadd__(self, other):
         """Return ``self+=value``."""
-        dpnp.add(self, other, out=self)
+        dpnp.add._inplace_op(self, other)
```

Comment:
This is how it was designed in dpctl. That PR is attempt to adopt to the API extension exposed by dpctl library.
Also `dpnp.add._inplace_op` path is much more simplier: some of the validation checks is not necessary there.
For sure, `dpnp.add._inplace_op` path can be subpart of `dpnp.add(a, b, out=a)` path in case when `out` refers to same memory as an input array, but I believe that has to be done by dpctl, not in dpnp.

Reply:
AFAIK `array api` doesn't support `out` parameter, so `dpctl` does not support it either. So it cannot be implemented inside `dpctl`.

I believe separating these two paths in `dpnp` is extremely bad idea. Which also requires more code changes than just making `add._inplace_op` as subpath of `add`.

Reply:
`dpctl` supports `out` keyword, but it's true that python array API doesn't.

Reply:
And what `numpy` do in this case?